src/libraries/System.Private.CoreLib/src/System/Reflection/InvokerEmitUtil.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection.Emit;
+
+namespace System.Reflection
+{
+    internal static class InvokerEmitUtil
+    {
+        // If changed, update native stack walking code that also uses this prefix to ignore reflection frames.
+        private const string InvokeStubPrefix = "InvokeStub_";
+
+        // This is an instance method to avoid overhead of shuffle thunk.
```

Comment:
This comment does not make sense on delegate.

Reply:
Delegate type is not a method. This comment says that this delegate type is an instance method, that does not make sense to me.

I am just saying that this comment should be somewhere else and/or reworded.

Reply:
I see. Will reword to something like
```cs
// The 'obj' parameter allows the DynamicMethod to be treated as an instance method which is slightly faster than a static.
```

Reply:
Have you stepped through the code to validate that there is no shuffle thunk? I think the way you have it setup will result into a shuffle thunk.

Reply:
No I haven't. I'll do that.

Reply:
The latest commit I believe fixes this. I verified that `StubLinkerCPU::EmitShuffleThunk` is no longer called. Any perf improvements are within the margin of error, but I did update the benchmark numbers. They are perhaps slightly better; there were also other minor refactoring due to the Mono ifdef work so we're not passing extra unused arguments anymore.